core/src/main/java/org/apache/iceberg/GenericPartitionFieldSummary.java

```diff
@@ -72,10 +73,11 @@ public GenericPartitionFieldSummary(Schema avroSchema) {
     }
   }
 
-  public GenericPartitionFieldSummary(boolean containsNull, ByteBuffer lowerBound,
+  public GenericPartitionFieldSummary(boolean containsNull, Boolean containsNaN, ByteBuffer lowerBound,
```

Comment:
Why use `Boolean` instead of the primitive here? Is there a place that constructs these without knowing whether there are `NaN` values?

Reply:
This is for a [test case](https://github.com/apache/iceberg/pull/1872/files#diff-366ef0b17670bc9547fe6d5b7b71a3d95c7f17eb97afa33c03dcdaeab27deb33R221) that provide null here to test backward compatibility. I guess I'll mark this constructor as `@VisibleForTesting` and create another public one that accepts `boolean`?